src/pages/index.js

```diff
@@ -5,96 +5,235 @@ import PopupWithImage from "../components/PopupWithImage.js";
 function handleImageClick(data) {
   cardPreview.open(data);
 }
 
-//Button consts
-const profileAddButton = document.querySelector(".profile__add-button");
-const profileEditButton = document.querySelector(".profile__edit-button");
-
-// button event listeners
-
 profileAddButton.addEventListener("click", () => {
   addPlacePopup.open();
 });
 
+editAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+  avatarFormValidator.disableButton();
+});
+
 profileEditButton.addEventListener("click", () => {
   profileEditPopup.open();
   editFormValidator.disableButton();
-  const currentUserInfo = userInfo.getUserInfo();
-  document.querySelector("#profile-name").value = currentUserInfo.name;
-  document.querySelector("#profile-description").value =
-    currentUserInfo.description;
+  api.getUserInfo().then((res) => {
+    document.querySelector("#profile-name").value = res.name;
+    document.querySelector("#profile-description").value = res.about;
+  });
 });
 
-// form submissions
+//feature functions
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteModal,
+    handleLikeCard
+  );
   const cardElement = card.getCardInfo();
   return cardElement;
 }
 
+function handleDeleteModal(card) {
+  deletePopup.setSubmitHandler(() => handleDeleteCard(card));
+  deletePopup.open();
+}
+
+function handleDeleteCard(card) {
+  api
+    .deleteCard(card.getCardId())
+    .then(() => {
+      card.handleDeleteCard();
+      deletePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked) {
+    api
+      .unlikeCard(card.getCardId())
+      .then((res) => {
+        card.handleLikeCard(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card.getCardId())
+      .then((res) => {
+        card.handleLikeCard(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+//all form submissions
+
 function handleAddCardFormSubmit(data) {
-  const newCard = createCard(data);
-  cardSection.addItem(newCard);
-  addPlacePopup.close();
+  addPlacePopup.renderloading(true);
+  api
+    .addCard(data)
+    .then((data) => {
+      const newCardElement = createCard(data);
+      cardSection.addItem(newCardElement);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addFormValidator.disableButton();
+      addPlacePopup.renderloading(false);
+      addPlacePopup.close();
+    });
   addPlacePopup.resetForm();
-  addFormValidator.disableButton();
 }
 
-function handleProfileFormSubmit(formValues) {
-  userInfo.setUserInfo(formValues.name, formValues.description);
-  profileEditPopup.close();
+function handleProfileFormSubmit(userData) {
+  profileEditPopup.renderloading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```